packages/react-native/src/primitives/ErrorMessage/__tests__/ErrorMessage.spec.tsx

```diff
@@ -1,34 +1,54 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { Text, View } from 'react-native';
 import { fireEvent, render } from '@testing-library/react-native';
+
+import { CLOSE_BUTTON_TEST_ID } from '../ErrorMessage';
 import ErrorMessage from '../ErrorMessage';
 
+const OnDismissExample = ({
+  customOnDismiss,
+}: {
+  customOnDismiss: jest.Mock<any, any>;
+}) => {
+  const [dismissed, setDismissed] = useState(false);
+
+  const handleOnDismiss = () => {
+    customOnDismiss();
+    setDismissed(!dismissed);
+  };
+
+  return (
+    <View>
+      {dismissed ? (
+        <Text>ErrorMessage dismissed</Text>
+      ) : (
+        <ErrorMessage onDismiss={handleOnDismiss}>
+          Test onDismiss handler
+        </ErrorMessage>
+      )}
+    </View>
+  );
+};
+
 describe('ErrorMessage', () => {
   it('renders default ErrorMessage as expected', () => {
-    const { toJSON } = render(
+    const { toJSON, queryByTestId } = render(
       <ErrorMessage>Default ErrorMessage</ErrorMessage>
     );
     expect(toJSON()).toMatchSnapshot();
+    expect(queryByTestId(CLOSE_BUTTON_TEST_ID)).toBeFalsy();
   });
 
   it('handles an onDismiss callback', () => {
     const customOnDismiss = jest.fn();
     const { queryByRole, getByTestId } = render(
-      <ErrorMessage onDismiss={customOnDismiss}>
-        Test onDismiss handler
-      </ErrorMessage>
+      <OnDismissExample customOnDismiss={customOnDismiss} />
```

Comment:
Could we just pass `customOnDismiss` to an `ErrorMessage` directly here?

Reply:
Yes, but we need a parent component to `ErrorMessage` in order to test that it gets removed from the screen

Reply:
Why? 😄

Reply:
Yeah good point 😂. Life is easier when it's a controlled component